packages/video_player/video_player_avfoundation/darwin/video_player_avfoundation/Sources/video_player_avfoundation/FVPVideoPlayerPlugin.m

```diff
@@ -431,11 +431,9 @@ - (void)setupEventSinkIfReadyToPlay {
     }
 
     BOOL hasVideoTracks = [asset tracksWithMediaType:AVMediaTypeVideo].count != 0;
-    BOOL hasNoTracks = asset.tracks.count == 0;
+    BOOL hasNoTracks = currentItem.tracks.count == 0 && asset.tracks.count == 0;
```

Comment:
can you explain why this is related to video vs audio tracks? (not quite sure how this fixed the audio only files)

Reply:
@hellohuanlin 
This is because HLS files AVAsset does not always have the track information because HLS is a streaming file, so we're checking `currentItem.tracks.count` to solve this problem.

Also, it is related to the size. HLS video files have height and width, but the HLS audio files do not. The differences affect the later if-statement, and that's why adding the checking above.

Reply:
So for audio only files, `currentItem.tracks` is not empty, but asset.tracks is empty?

Reply:
Yes, that's right! 🙂

Reply:
This kind of stuff needs to be commented since it's not apparent at first. Could you add some comment? Thank you!

Reply:
Added some comments in the below commit. Tell me a better comment if you have:
[Add clarification comment for checking HLS audio-only tracks](https://github.com/flutter/packages/pull/7890/commits/ac6c52859092a5174eff60c1bb7e2579c64f06db)